modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/EditAPIApplicationEndPoint.testcase

```diff
@@ -0,0 +1,281 @@
+			SchemaAPI.createAPISchema(
+				externalReferenceCode = "My-app",
+				mainObjectDefinitionERC = "university",
+				name = "testSchema");
+
+			APIBuilderUI.addAndPublishSchemaProperties(
+				entity = "Schemas",
+				fieldName = "name",
+				name = "testSchema",
+				propertyName = "universityName",
+				relatedFieldName = "name",
+				relatedObjectName = "student",
+				relatedPropertyName = "studentName",
+				title = "My-app");
+		}
+
+		task ("And Given Schema related to the endpoint") {
+			APIBuilderUI.addAndPublishEndpointConfiguration(
+				entity = "Endpoints",
+				name = "testendpoint",
+				schemaName = "testSchema",
+				title = "My-app");
+		}
+
+		task ("When executing the endpoint") {
+			APIExplorer.navigateToOpenAPI(customObjectPlural = "my-app");
+
+			APIExplorer.executeAPIMethod(
+				method = "getTestendpointPage",
+				service = "testSchema");
+		}
+
+		task ("Then all fields are correctly returned") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "OpenAPI#RESPONSE_BODY",
+				method = "getTestendpointPage",
+				value1 = "\"items\": [     {       \"universityName\": \"Oxford\",       \"studentName\": [         \"Pearl\"       ]     }   ]");
+		}
+	}
+
+	@priority = 4
+	test CanRelateSchemaWithManyEndpoints {
+		property portal.acceptance = "true";
+
+		task ("Given API Application with two endpoints created") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				status = "published",
+				title = "My-app");
+
+			EndpointAPI.createNAPIEndpoints(
+				externalReferenceCode = "My-app",
+				name = "testendpoint",
+				numberOfEndpoints = 2,
+				path = "/testendpoint");
+		}
+
+		task ("And Given one API Schema in API Application") {
+			SchemaAPI.createAPISchema(
+				externalReferenceCode = "My-app",
+				mainObjectDefinitionERC = "L_API_APPLICATION",
+				name = "testSchema");
+		}
+
+		task ("And Given modifying endpoint1 with API Schema") {
+			APIBuilderUI.addAndPublishEndpointConfiguration(
+				entity = "Endpoints",
+				name = "testendpoint1",
+				schemaName = "testSchema",
+				title = "My-app");
+		}
+
+		task ("When modifying endpoint2 wit the same API Schema") {
+			APIBuilderUI.addAndPublishEndpointConfiguration(
+				entity = "Endpoints",
+				name = "testendpoint2",
+				schemaName = "testSchema",
+				title = "My-app");
+		}
+
+		task ("Then both endpoints work with the same schema") {
+			var response = EndpointAPI.getAPIEndpoints();
+
+			var actualValue = JSONPathUtil.getProperty(
+				property = "$.items..r_responseAPISchemaToAPIEndpoints_c_apiSchemaERC",
+				response = ${response});
+
+			TestUtils.assertEquals(
+				actual = ${actualValue},
+				expected = "testSchema,testSchema");
+		}
+	}
+
+	@priority = 4
+	test CanUpdateEndpoint {
+		property portal.acceptance = "true";
+
+		task ("Given API Application with endpoint created") {
+			ApplicationAPI.createAPIApplication(
```

Comment:
I think we can move createAPIApplication into setup since every case need to create it.